From e413c4c97b7acfc2d3814a87ca68e6f47a432a36 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Wed, 1 Jun 2022 16:14:17 +0800
Subject: [PATCH 26/31] Revert "drm/amd/display: Add affected crtcs to atomic
 state for dsc mst unplug"

This reverts commit 2eaa9d86e00713587635c7c9d9b4576d746ae101.
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index c5231c50c412..29f8b4bd5a76 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -7268,13 +7268,10 @@ dm_determine_update_type_for_commit(struct amdgpu_display_manager *dm,
 static int add_affected_mst_dsc_crtcs(struct drm_atomic_state *state, struct drm_crtc *crtc)
 {
 	struct drm_connector *connector;
-	struct drm_connector_state *conn_state, *old_conn_state;
+	struct drm_connector_state *conn_state;
 	struct amdgpu_dm_connector *aconnector = NULL;
 	int i;
-	for_each_oldnew_connector_in_state(state, connector, old_conn_state, conn_state, i) {
-		if (!conn_state->crtc)
-			conn_state = old_conn_state;
-
+	for_each_new_connector_in_state(state, connector, conn_state, i) {
 		if (conn_state->crtc != crtc)
 			continue;
 
-- 
2.27.0

